{
  "cells": [
    {
      "cell_type": "raw",
      "metadata": {},
      "source": [
        "---\n",
        "title: P708 - stranded mRNA-Seq analysis of Human fibroblast skin model\n",
        "author:\n",
        "  - name: Assa Yeroslaviz\n",
        "    orcid: 0000-0001-9638-4026\n",
        "    corresponding: true\n",
        "    email: yeroslaviz@biochem.mpg.de\n",
        "    roles:\n",
        "      - Investigation\n",
        "      - Visualization\n",
        "    affiliations:\n",
        "      - Max-Planck-Institute for Biochemistry\n",
        "keywords:\n",
        "  - MPI. MR\n",
        "  - Dept. Spatz\n",
        "  - RNA-Seq\n",
        "  - Rahul Rimal\n",
        "  - Spheroids\n",
        "  - human fibroblast\n",
        "abstract: |\n",
        "  On June 2023 Rahul Rimal contacted Rinho and myself regarding the analysis of vascularized spheroids. The project contains two experimental conditions: a) Control Vascularized spheroids and b) Vascularized spheroids with Microgels. It was observed that addition of Microgels improves vascularization significantly. They would like to confirm this behavior based on the visual results acquired with sequencing data. \n",
        "plain-language-summary: |\n",
        "  RNA-Seq analysis with the goal of mapping and quantification of the two conditions\n",
        "key-points:\n",
        "  - \n",
        "date: last-modified\n",
        "bibliography: references.bib\n",
        "citation:\n",
        "  container-title: Spheroid Analysis\n",
        "number-sections: true\n",
        "format: \n",
        "  html:\n",
        "    code-fold: true\n",
        "---"
      ],
      "id": "7772198e"
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "## Introduction\n",
        "\n",
        "In the First email Rahul send he wrote:\n",
        "\n",
        "```         \n",
        "For my project, I am also interested in conducting the sequencing and would be grateful to have your expertise. I am currently working with vascularized spheroids and have two experimental conditions:\n",
        "\n",
        "a) Control Vascularized spheroids \n",
        "b) Vascularized spheroids with Microgels \n",
        "\n",
        "We have observed that addition of Microgels improves vascularization significantly and would like to confirm the visual results we acquired with the sequencing data. \n",
        " \n",
        "I am in the process of preparing/extracting RNA from 3 technical and 3 biological replicates of each above mentioned conditions. Therefore, a total of 18 samples. \n",
        "Each sample contain 300 vascularized spheroids (roughly a million cells per sample). \n",
        "\n",
        "I would like to know whether sequencing of these samples is feasible for you and your schedule. Do let me know which month/date could be the most suitable for you to receive the samples (depending on your schedule).\n",
        "```\n",
        "\n",
        "So, we have a 3x3 experimental design for two conditions.\n",
        "\n",
        "In a followed up mail a few weeks later he asked for:\n",
        "\n",
        "```         \n",
        "Fastq files are available. https://datashare.biochem.mpg.de/s/NMy4gBMQYzwTth1\n",
        "Could you please help me with this? \n",
        "\n",
        "I would require for now \n",
        "1) The PCA graph\n",
        "2) The volcano plot \n",
        "3) gene ontology graph\n",
        "\n",
        "Could you also provide me the excel sheet with the final gene expression values? \n",
        "Eagerly waiting for the results. \n",
        "```\n",
        "\n",
        "# Preparation\n",
        "\n",
        "The raw data is saved on the pool folder `pool-bcfngs` and was linked to my working folder.\n",
        "\n",
        "\n",
        "```{bash rawData}\n",
        "#| echo: true\n",
        "#| eval: false\n",
        "mkdir -p /fs/home/yeroslaviz/projects/MPI_MR/Rahul/P708/rawData\n",
        "cd rawData/\n",
        "ln -s /home/yeroslaviz/poolFolders/pool-bcfngs/fastq_files/P708/P708_20230808_Spatz_Rahul/conc.fastq/*. .\n",
        "```\n",
        "\n",
        "\n",
        "# Quality Control\n",
        "\n",
        "Quality control was done using the `fastqc` tool [@fastqc, v. 0.11.7] within the automated sequencing pipeline. QC looks good for all samples.\n",
        "\n",
        "generating a report using `multiQC` [@multiqc, v. 1.14] summarizing the QC Results.\n",
        "\n",
        "\n",
        "```{bash}\n",
        "#| echo: true\n",
        "#| eval: false\n",
        "conda activate Mapping\n",
        "\n",
        "mkdir ../fastqcResults\n",
        "cd  ../\n",
        "fastqc -t 22 -o fastqcResults/ rawData/*.gz\n",
        "\n",
        "multiqc -o QC /fs/home/yeroslaviz/projects/MPI_MR/Rahul/P708/fastqcResults/\n",
        "```\n",
        "\n",
        "\n",
        "## Mapping\n",
        "\n",
        "Mapping the samples was done using the STAR Aligner [@star, v. 2.7.10b]\n",
        "\n",
        "\n",
        "```{bash star}\n",
        "#| echo: true\n",
        "#| eval: false\n",
        "#| \n",
        "basisDir=\"/fs/home/yeroslaviz/projects/MPI_MR/Rahul/P708\"\n",
        "rawData=\"${basisDir}/rawData\"\n",
        "output=\"${basisDir}/STAR_mapping\"\n",
        "mkdir -p $output\n",
        "starIndex=\"/home/yeroslaviz/poolFolders/pool-bcfngs/genomes/Hsp.GRCh38/starIndex2.7.10a\"\n",
        "gtf=\"/home/yeroslaviz/poolFolders/pool-bcfngs/genomes/Hsp.GRCh38.gtf\"\n",
        "\n",
        "for file in `ls $rawData/*R1.fastq.gz`\n",
        "do\n",
        "    base=$(echo $file | sed -E \"s/\\/fs\\/home\\/yeroslaviz\\/projects\\/MPI_MR\\/Rahul\\/P708\\/rawData\\/(.*).conc.R1.fastq.gz/\\1/\")\n",
        "    echo -e \"\\n Mapping $base \\n\"\n",
        "    mkdir -p ${output}/$base/\n",
        "#    echo ${base}\n",
        "    STAR --runThreadN 20 --genomeDir $starIndex --sjdbGTFfile $gtf --sjdbOverhang 100 \\\n",
        "    --readFilesCommand zcat \\\n",
        "    --readFilesIn ${rawData}/$base.conc.R1.fastq.gz ${rawData}/$base.conc.R2.fastq.gz \\\n",
        "    --outFileNamePrefix ${output}/$base/ \\\n",
        "    --quantMode TranscriptomeSAM GeneCounts \\\n",
        "    --outSAMtype BAM SortedByCoordinate\n",
        "    mv ${output}/${base}/Aligned.sortedByCoord.out.bam ${output}/${base}.sorted.bam\n",
        "    samtools index ${output}/$base.sorted.bam\n",
        "    rm -rf ${output}/${base}/_STAR*\n",
        "done\n",
        "\n",
        "conda deactivate\n",
        "```\n",
        "\n",
        "\n",
        "| Sample Name    | \\# read counts | \\# uniq Mapped | \\% uniq Mapped | \\# Multi-Mapped | \\% Multi-Mapped | TotalMapped |\n",
        "|-----------|-----------|-----------|-----------|-----------|-----------|-----------|\n",
        "| control_r1_1   | 14934832       | 13903988       | 93.10%         | 764217          | 5.117%          | 98.217%     |\n",
        "| control_r1_2   | 16851257       | 15957162       | 94.69%         | 652037          | 3.869%          | 98.559%     |\n",
        "| control_r1_3   | 17756996       | 16893231       | 95.14%         | 680413          | 3.832%          | 98.972%     |\n",
        "| control_r2_1   | 19058058       | 18107861       | 95.01%         | 778372          | 4.084%          | 99.094%     |\n",
        "| control_r2_2   | 17399089       | 16502506       | 94.85%         | 700218          | 4.024%          | 98.874%     |\n",
        "| control_r2_3   | 9044369        | 8510562        | 94.10%         | 369782          | 4.089%          | 98.189%     |\n",
        "| control_r3_1   | 20002695       | 19052374       | 95.25%         | 715315          | 3.576%          | 98.826%     |\n",
        "| control_r3_2   | 14401432       | 13746894       | 95.46%         | 503051          | 3.493%          | 98.953%     |\n",
        "| control_r3_3   | 23793484       | 22452657       | 94.36%         | 884989          | 3.719%          | 98.079%     |\n",
        "| microgels_r1_1 | 14394900       | 13739503       | 95.45%         | 530096          | 3.683%          | 99.133%     |\n",
        "| microgels_r1_2 | 19167758       | 18295504       | 95.45%         | 705631          | 3.681%          | 99.131%     |\n",
        "| microgels_r1_3 | 17222387       | 16408266       | 95.27%         | 650330          | 3.776%          | 99.046%     |\n",
        "| microgels_r2_1 | 18897903       | 18021886       | 95.36%         | 722859          | 3.825%          | 99.185%     |\n",
        "| microgels_r2_2 | 14837465       | 14155321       | 95.40%         | 553967          | 3.734%          | 99.134%     |\n",
        "| microgels_r2_3 | 14117159       | 13446885       | 95.25%         | 542432          | 3.842%          | 99.092%     |\n",
        "| microgels_r3_1 | 17627394       | 16817864       | 95.41%         | 673610          | 3.821%          | 99.231%     |\n",
        "| microgels_r3_2 | 14383836       | 13711139       | 95.32%         | 530871          | 3.691%          | 99.011%     |\n",
        "| microgels_r3_3 | 12048255       | 11495761       | 95.41%         | 453869          | 3.767%          | 99.177%     |\n",
        "|                |                |                |                |                 |                 |             |\n",
        "\n",
        "## Data & Methods {#sec-data-methods}\n",
        "\n",
        "## Conclusion\n",
        "\n",
        "## References {.unnumbered}\n",
        "\n",
        "::: {#refs}\n",
        ":::"
      ],
      "id": "80f5bd96"
    }
  ],
  "metadata": {
    "kernelspec": {
      "name": "env",
      "language": "python",
      "display_name": "Environment (env)"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 5
}